libpod/container_internal.go

```diff
@@ -671,31 +685,32 @@ func (c *Container) mountStorage() (err error) {
 		return nil
 	}
 
-	// TODO: generalize this mount code so it will mount every mount in ctr.config.Mounts
-
-	mounted, err := mount.Mounted(c.config.ShmDir)
-	if err != nil {
-		return errors.Wrapf(err, "unable to determine if %q is mounted", c.config.ShmDir)
-	}
-
-	if err := os.Chown(c.config.ShmDir, c.RootUID(), c.RootGID()); err != nil {
-		return err
-	}
-
-	if !mounted {
-		shmOptions := fmt.Sprintf("mode=1777,size=%d", c.config.ShmSize)
-		if err := unix.Mount("shm", c.config.ShmDir, "tmpfs", unix.MS_NOEXEC|unix.MS_NOSUID|unix.MS_NODEV,
-			label.FormatMountLabel(shmOptions, c.config.MountLabel)); err != nil {
-			return errors.Wrapf(err, "failed to mount shm tmpfs %q", c.config.ShmDir)
+	if os.Getuid() == 0 {
+		// TODO: generalize this mount code so it will mount every mount in ctr.config.Mounts
+		mounted, err := mount.Mounted(c.config.ShmDir)
+		if err != nil {
+			return errors.Wrapf(err, "unable to determine if %q is mounted", c.config.ShmDir)
 		}
+
 		if err := os.Chown(c.config.ShmDir, c.RootUID(), c.RootGID()); err != nil {
```

Comment:
Before this PR, there were two `Chown(c.config.ShmDir` calls (one before and one after the `if !mounted`).  Was the second one removed intentionally?

Reply:
Still outstanding

Reply:
reintroduced